tools/update_versions_readme.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (C) IBM Corp. 2016.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os
+import shutil
+
+from lxml import etree
+import git
+
+from lib import config
+from lib import distro_utils
+from lib import exception
+from lib import packages_manager
+from lib import repository
+from lib import rpm_package
+from lib.utils import replace_str_in_file
+from lib.versions_repository import setup_versions_repository
+
+
+LOG = logging.getLogger(__name__)
+
+
+def generate_html_table(packages):
+    table = etree.Element('table')
+    thead = etree.SubElement(table, 'thead')
+    software_th = etree.SubElement(thead, 'th')
+    software_th.text = 'Software'
+
+    version_th = etree.SubElement(thead, 'th')
+    version_th.text = 'Version'
+
+    tbody = etree.SubElement(table, 'tbody')
+
+    for package in packages:
+        tr = etree.SubElement(tbody, 'tr')
+        package_name_td = etree.SubElement(tr, 'td')
+        package_name_td.text = package.name
+
+        package_version_td = etree.SubElement(tr, 'td')
+        package_version_td.text = package.version
+
+    return etree.tostring(table, pretty_print=True)
+
+
+def update_versions_in_readme(versions_repo, distro, packages):
+    """
+    Update packages versions in README
+
+    Args:
+        versions_repo (GitRepository): versions git repository handler
+        distro (distro.LinuxDistribution): Linux distribution
+        packages ([str]): list of packages which versions must be updated
```

Comment:
I'd name this variable `package_names` to avoid confusing with the `Package` type.